jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Rabit.java

```diff
@@ -92,4 +131,73 @@ public static int getWorldSize() throws XGBoostError {
     checkCall(XGBoostJNI.RabitGetWorldSize(out));
     return out[0];
   }
+
+  public static float[] allReduce(float[] elements, OpType op) {
```

Comment:
That was purely for the unit test actually. I wanted to test the Rabit tracker minimally possible. Originally I wanted to write the test cases in `xgboost4j`, but somehow `allReduce` can't be called in separate threads so I ended up moving the unit tests to `xgboost4j-spark`, utilizing Spark's executor/container to handle multiple processes for me.